### DISCUSSION TITLE
Prettier Extra Plugin not working as expected

### DISCUSSION THREAD

**Discussion by madflow** [LazyVim/LazyVim#1119] [Q&A]
In a Next.JS project with TS, the prettier formatting "suddenly" behaved differently than before, and breaks teh formatting on save for me.

I have to save twice, so the prettier formatting will work.

I tried to debug this by creating a new Next.JS project and using a fresh LazyVim Installation, with just 

```lua
require("lazy").setup({
  spec = {
    -- add LazyVim and import its plugins
    { "LazyVim/LazyVim", import = "lazyvim.plugins" },
    -- import any extras modules here
    -- { import = "lazyvim.plugins.extras.lang.typescript" },
    -- { import = "lazyvim.plugins.extras.lang.json" },
    -- { import = "lazyvim.plugins.extras.ui.mini-animate" },
    { import = "lazyvim.plugins.extras.formatting.prettier" },
    -- import/override with your plugins
    { import = "plugins" },
```

I created a `prettier.config.js` with just:

```js
module.exports = {
  semi: true,
  singleQuote: true,
};
```

However - when I save a tsx file - the prettier config is not applied.  

Note, that the "singlequote" is not applied.

![Kapture 2023-07-13 at 19 04 18](https://github.com/LazyVim/LazyVim/assets/183248/06d27b4f-014a-4d8b-adc2-bff09db017bc)

Is there something I can do?

**Comment by folke** [comment]
What does `:NullLsInfo` show?

Most likely you are overriding null-ls `opts.sources` in your own config.

If that's the case, you need to change it to something like:
```lua
  {
    "jose-elias-alvarez/null-ls.nvim",
    -- dev = true,
    opts = function(_, opts)
      local nls = require("null-ls")
      vim.list_extend(opts.sources, {
-- add your sources here

**Reply by madflow** [reply]
Thanks, I forgot sbout `:NullLsInfo`. I rebooted my Macbook (even though people tell me, to never do that :) - now it is working again.

I suspect, that the `prettierd` process was running in the background in a defunct state. I was working in an older codebase before - maybe it started with an old Node Version or sth.

This is a Ecosystem problem I would argue and not from LazyVim. So closing here.
